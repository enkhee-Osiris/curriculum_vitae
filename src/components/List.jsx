import { v4 as uuidv4 } from 'uuid';

function List({ items }) {
  return (
    <ul className="space-y-1 list-disc list-inside">
      {items.map((item) => (
        <li key={uuidv4()}>{item}</li>
      ))}
    </ul>
  );
}

export default List;
