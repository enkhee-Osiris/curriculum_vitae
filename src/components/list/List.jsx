import React from 'react';
import { arrayOf, string } from 'prop-types';

const List = ({ items }) => {
  /* eslint-disable react/no-array-index-key */
  const listItems = items.map((item, index) => (
    <li key={index}>
      {item}
    </li>));

  return <ul>{listItems}</ul>;
};

List.propTypes = {
  items: arrayOf(string).isRequired,
};

export { List as default };
