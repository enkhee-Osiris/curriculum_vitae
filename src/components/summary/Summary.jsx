import React, { Fragment } from 'react';
import { arrayOf, string } from 'prop-types';
import List from '../list/List';

const Summary = ({ items }) => (
  <Fragment>
    <h3>Summary</h3>
    <List items={items} />
  </Fragment>
);

Summary.propTypes = {
  items: arrayOf(string).isRequired,
};

export { Summary as default };
