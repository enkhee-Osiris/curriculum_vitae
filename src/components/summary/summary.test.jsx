import React from 'react';
import { shallow } from 'enzyme';
import Summary from './Summary';
import List from '../list/List';

describe('<Summary />', () => {
  const props = {
    items: ['test'],
  };

  it('should be defined', () => {
    expect(Summary).toBeDefined();
  });

  it('renders correctly', () => {
    const summary = shallow(<Summary {...props} />);

    expect(summary).toMatchSnapshot();
  });

  it('renders h3', () => {
    const summary = shallow(<Summary {...props} />);
    const title = <h3>Summary</h3>;

    expect(summary.contains(title)).toEqual(true);
  });

  it('renders List', () => {
    const summary = shallow(<Summary {...props} />);
    const list = <List items={props.items} />;

    expect(summary.contains(list)).toEqual(true);
  });
});
