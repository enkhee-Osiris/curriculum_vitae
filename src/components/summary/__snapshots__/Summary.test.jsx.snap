// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Summary
    items={
        Array [
            "test",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Summary
</h3>,
        <List
          items={
                    Array [
                              "test",
                            ]
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Summary",
        },
        "ref": null,
        "rendered": "Summary",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "items": Array [
            "test",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Summary
</h3>,
          <List
            items={
                        Array [
                                    "test",
                                  ]
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Summary",
          },
          "ref": null,
          "rendered": "Summary",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "items": Array [
              "test",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
